code/modules/mob/living/carbon/alien/alien.dm

```diff
@@ -295,3 +295,30 @@ Des: Removes all infected images from the alien.
 #undef HEAT_DAMAGE_LEVEL_1
 #undef HEAT_DAMAGE_LEVEL_2
 #undef HEAT_DAMAGE_LEVEL_3
+
+/mob/living/carbon/alien/handle_footstep(turf/T)
+	if(..())
+		if(T.footstep_sounds["xeno"])
+			var/S = pick(T.footstep_sounds["xeno"])
+			if(S)
+				if(m_intent == "run")
+					if(!(step_count % 2)) //every other turf makes a sound
+						return 0
+
+				var/range = -(world.view - 2)
+				range -= 0.666 //-(7 - 2) = (-5) = -5 | -5 - (0.666) = -5.666 | (7 + -5.666) = 1.334 | 1.334 * 3 = 4.002 | range(4.002) = range(4)
+				var/volume = 5
+
+				if(m_intent == "walk")
+					return 0 //silent when walking
+
+				if(buckled || lying || throwing)
+					return 0 //people flying, lying down or sitting do not step
+
+				if(!has_gravity(src))
+					if(step_count % 3) //this basically says, every three moves make a noise

```

Comment:
Not what this says at all.
1 mod 3 is 1
2 mod 3 is 2
3 mod 3 is 0

```if(step_count % 3 == 0)``` is "every 3 steps"